.github/workflows/build-container.yml

```diff
@@ -22,63 +99,140 @@ jobs:
       GPG_KEY_PASSWORD: ${{ secrets.GPG_KEY_PASSWORD }}
       DOCKER_HUB_ACCESS_TOKEN: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
       DOCKER_HUB_USER: ${{ secrets.DOCKER_HUB_USER }}
+      LOCAL_REPO: /maven
     steps:
       - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
 
-      - name: Install deps
+      - name: Work around git permission issue
         run: |
-          apk add libxml2-utils docker
+          dname=$(echo ${{github.repository}} | cut -d'/' -f2)
+          git config --global --add safe.directory /__w/$dname/$dname
+        shell: sh
 
-      - name: Packer Version
+      - name: Build repository
+        id: build
         run: |
-          packer version
+          # We have just checked out the top HEAD,
+          # so save the build script separately
+          # to use for building everything
+          cp for-ci-build-image/maven-build.sh .
```

Comment:
todo: Unless I am mistaken this is copying the script to the a directory that is under git control, so the hard reset will remove it if you process more than a single commit.

If this is the case, this needs to be updated, otherwise can you please add an explanation to describe how this is not the case

Reply:
I've double-checked. `git reset --hard` will only remove the changes in the tracked files if there are any, or the actual untracked files if they're in a way of a tracked file.

Considering we're creating a new file in the root of the project (`maven-build.sh`), which is not in a way of anything else, this file will indeed persist through all the git reset commands as it's not tracked.